src/main/scala/tech/cryptonomic/conseil/metadata/MetadataService.scala

```diff
@@ -0,0 +1,63 @@
+package tech.cryptonomic.conseil.metadata
+
+import tech.cryptonomic.conseil.config.MetadataOverridesConfiguration
+import tech.cryptonomic.conseil.config.Platforms.PlatformsConfiguration
+import tech.cryptonomic.conseil.generic.chain.PlatformDiscoveryTypes.{Attribute, Entity, Network, Platform}
+import tech.cryptonomic.conseil.tezos.TezosPlatformDiscoveryOperations
+import tech.cryptonomic.conseil.util.CollectionOps.{ExtendedFuture, ExtendedFutureWithOption}
+import tech.cryptonomic.conseil.util.ConfigUtil
+import tech.cryptonomic.conseil.util.OptionUtil.when
+
+import scala.concurrent.Future.successful
+import scala.concurrent.{ExecutionContext, Future}
+
+class MetadataService(config: PlatformsConfiguration,
+                      metadataOverrides: MetadataOverridesConfiguration,
+                      tezosPlatformDiscoveryOperations: TezosPlatformDiscoveryOperations) {
+
+  val overridePlatform = UnitTransformation.platform(metadataOverrides)
+  val overrideNetwork = UnitTransformation.network(metadataOverrides)
+  val overrideEntity = UnitTransformation.entity(metadataOverrides)
+  val overrideAttribute = UnitTransformation.attribute(metadataOverrides)
+
+  def getPlatforms: List[Platform] = ConfigUtil
+    .getPlatforms(config)
+    .flatMap(platform => overridePlatform(platform, PlatformPath(platform.name)))
+
+  def getNetworks(path: PlatformPath): Option[List[Network]] = when(metadataOverrides.isVisible(path)) {
+    ConfigUtil
+      .getNetworks(config, path.platform)
+      .flatMap(network => overrideNetwork(network, path.addLevel(network.name)))
+  }
+
+  def getEntities(path: NetworkPath)(implicit apiExecutionContext: ExecutionContext): Future[Option[List[Entity]]] = {
+    if (exists(path) && metadataOverrides.isVisible(path))
+      tezosPlatformDiscoveryOperations
+        .getEntities
+        .embeddedMap(entity => overrideEntity(entity, path.addLevel(entity.name)))
+        .map(Some(_))
+    else
+      successful(None)
```

Comment:
This is a constant value, you might want to store it as a named `val` e re-use it, with the additional possible side-effect of improving readability, too?